lib/usd/translators/shading/usdFileTextureWriter.cpp

```diff
@@ -452,6 +472,178 @@ void PxrUsdTranslators_FileTextureWriter::Write(const UsdTimeCode& usdTime)
+    }
+
+    // Scale
+    const MPlug repeatUVPlug = depNodeFn.findPlug(
+        _tokens->repeatUV.GetText(),
+        /* wantNetworkedPlug = */ true,
+        &status);
+    if (status != MS::kSuccess) {
+        return;
+    }
+
+    if (UsdMayaUtil::IsAuthored(offsetPlug)) {
+        for (size_t i = 0u; i < GfVec2f::dimension; ++i) {
+            scaleValue[i] = repeatUVPlug.child(i).asFloat(&status);
+            if (status != MS::kSuccess) {
+                return;
+            }
+        }
+
+        if (scaleValue != scaleFallbackValue) {
+            transformationsAreAuthored |= true;
+        }
+    }
+
+    if (!transformationsAreAuthored) {
+        return;
+    }
+
+    // Get the TexCoordReader node and its output "result"
+    const SdfPath primvarReaderShaderPath
+        = texShaderSchema.GetPath().AppendChild(_tokens->PrimvarReaderShaderName);
+
+    const UsdShadeShader primvarReaderShader
+        = texShaderSchema.Get(GetUsdStage(), primvarReaderShaderPath);
+
+    const UsdShadeOutput primvarReaderShaderOutput = primvarReaderShader.GetOutput(_tokens->result);
+
+    // Create the Transform2d node as a child of the UsdUVTexture node
+    const SdfPath transform2dShaderPath
+        = texShaderSchema.GetPath().AppendChild(_tokens->UsdTransform2dShaderName);
+    UsdShadeShader transform2dShaderSchema
+        = UsdShadeShader::Define(GetUsdStage(), transform2dShaderPath);
+
+    transform2dShaderSchema.CreateIdAttr(VtValue(_tokens->UsdTransform2d));
+
+    // Create the Transform2d input "in" attribute and connect it
+    // to the TexCoordReader output "result"
+    transform2dShaderSchema.CreateInput(_tokens->in, SdfValueTypeNames->Float2)
+        .ConnectToSource(primvarReaderShaderOutput);
+
+    // Compute the Transform2d values, converting from Maya's coordinates to USD coordinates
+
+    // Maya's place2dtexture transform order seems to be `in * T * S * R`, where the rotation pivot
+    // is (0.5, 0.5) and scale pivot is (0,0). USD's Transform2d transform order is `in * S * R *
+    // T`, where the rotation and scale pivots are (0,0). This conversion translates from
+    // place2dtexture's UV space to Transform2d's UV space: `in * S * T * Rpivot_inverse * R *
+    // Rpivot`
+    GfMatrix4f pivotXform = GfMatrix4f().SetTranslate(GfVec3f(0.5, 0.5, 0));
+    GfMatrix4f translateXform
+        = GfMatrix4f().SetTranslate(GfVec3f(translationValue[0], translationValue[1], 0));
+    GfRotation rotation = GfRotation(GfVec3f::ZAxis(), rotationValue);
+    GfMatrix4f rotationXform = GfMatrix4f().SetRotate(rotation);
+    GfVec3f    scale;
+    if (fabs(scaleValue[0]) <= std::numeric_limits<float>::epsilon()
+        || fabs(scaleValue[1]) <= std::numeric_limits<float>::epsilon()) {
+        TF_WARN(
+            "At least one of the components of RepeatUV are set to zero.  To avoid divide by zero "
+            "exceptions, these values are changed to the smallest finite float greater than zero.");
```

Comment:
Might be nice to include the full name of the problematic node in this message to help track it down. You can get that with this:

```C++
UsdMayaUtil::GetMayaNodeName(GetMayaObject())
```

Reply:
Good call, I've swapped that out.